tripy/tripy/backend/compiler_api.py

```diff
@@ -33,6 +33,97 @@
 from tripy.utils import json as json_utils
 
 
+@export.public_api(document_under="compiler")
+class Stream:
+    """
+    Represents CUDA streams that can be used to manage concurrent operations.
+
+    This class is a wrapper around the underlying stream object, allowing management of CUDA streams.
+
+    Args:
+        create_new (bool): If True, creates a new stream. If False, uses the default stream.
+
+    .. code-block:: python
+        :linenos:
+        :caption: Create new streams
+
+        streamA = tp.Stream()
+        streamB = tp.Stream()
+        defaultStream = tp.Stream(create_new=False)
+        assert streamA != streamB
+        assert defaultStream != streamB
+
+    .. code-block:: python
+        :linenos:
+        :caption: Use streams in execution
+
+        linear = tp.Linear(2, 3)
+        compiler = tp.Compiler(linear)
+
+        compiled_linear = compiler.compile(tp.InputInfo((2, 2), dtype=tp.float32))
+
+        a = tp.ones((2, 2), dtype=tp.float32)
+
+        compiled_linear.stream = tp.Stream()
```

Comment:
Can streams not be used in eager mode? I think using a context manager to set the current stream and making the behavior consistent in both eager/compiled mode would be preferable

Reply:
Updated to make Stream work with context. @jhalakpatel  @yizhuoz004 can you please re-review.